src/NetworkManager.cpp

```diff
@@ -0,0 +1,469 @@
+#include "global.h"
+#include "NetworkManager.h"
+#include "LuaManager.h"
+#include "RageLog.h"
+#include "RageUtil.h"
+#include "StdString.h"
+
+#include <ixwebsocket/IXHttpClient.h>
+#include <ixwebsocket/IXNetSystem.h>
+#include <ixwebsocket/IXUrlParser.h>
+
+#include <algorithm>
+#include <unordered_map>
+#include <vector>
+
+
+NetworkManager*	NETWORK = nullptr;	// global and accessible from anywhere in our program
+
+Preference<bool> NetworkManager::httpEnable("HttpEnable", false);
+Preference<RString> NetworkManager::httpAllowHosts("HttpAllowHosts", "");
```

Comment:
Nothing to change here, but as a heads up I intend to make this a "Read only" preference (once I add in that mechanism) so that you can't change it with PREFSMAN:SetPreference().